### ISSUE TITLE
JSONDecodeError on Python 3.10

### ISSUE THREAD

**Issue by robd003** [python-poetry/poetry#4210]
- [X] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [X] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [X] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS 11.4
- **Poetry version**: 1.1.6
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/robd003/9147c6ec95678bd578b6da828f1b3840

## Issue
Poetry crashes whenever it evaluates a package for installation. Running `poetry install` will return JSONDecodeError 100% of the time.

```
$ poetry install
Installing dependencies from lock file

Package operations: 16 installs, 0 updates, 0 removals

  • Installing sanic-routing (0.6.2): Failed

  JSONDecodeError

  Expecting value: line 1 column 1 (char 0)

  at /usr/local/Cellar/python@3.10/3.10.0b3/Frameworks/Python.framework/Versions/3.10/lib/python3.10/json/decoder.py:355 in raw_decode
      351│         """
      352│         try:
      353│             obj, end = self.scan_once(s, idx)
      354│         except StopIteration as err:
    → 355│             raise JSONDecodeError("Expecting value", s, err.value) from None
      356│         return obj, end
      357│ 
```

**Comment by robd003** [comment]
I've tried removing the entire pypoetry cache directory and still having the same issue

**Comment by robd003** [comment]
Just tried the 1.2.0a1 release and it's working fine.

**Comment by robd003** [comment]
The 1.1.7 update pushed out today still has the same problem on Python 3.10

**Comment by hoefling** [comment]
Reproducible with 3.10 on Linux & Windows as well.

**Comment by hoefling** [comment]
BTW disabling `poetry`'s experimental new installer may be a workaround for now:
```sh
$ poetry config experimental.new-installer false
```

**Comment by seandstewart** [comment]
I can confirm that the issue is resolved by @hoefling's suggestion.

> BTW disabling `poetry`'s experimental new installer may be a workaround for now:
> 
> ```shell
> $ poetry config experimental.new-installer false
> ```

**Comment by dmerejkowsky** [comment]
I can confirm I can reproduce here. Poke @python-poetry/triage

**Comment by dmerejkowsky** [comment]
And this is fixed in 1.2.0a2. I guess we can close this ?

**Comment by remram44** [comment]
I am glad I found this issue. I wouldn't have known how to even debug this, since **Poetry eats the whole traceback I only shows this JSONDecodeError**.

Is there a way to print the traceback?

**Comment by BSVogler** [comment]
Anyone got this on 3.9.7? I get

```
 • Installing pycparser (2.20): Failed

  JSONDecodeError

  Expecting value: line 1 column 1 (char 0)

  at /usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/decoder.py:355 in raw_decode
```

**Comment by ksamuel** [comment]
Same here, with a project  with both 3.9 and 3.10 env. Removing the 3.10 env solve the problem. However, 3.10 will be released in 2 days, so I don't think we should be surprise there are still things that don't work yet.

**Comment by jodal** [comment]
Reproduced with Poetry 1.1.10 installed on Python 3.9.5 (Ubuntu 21.04 system Python) using pipx.

`poetry install` started failing immediately when I switched project from Python 3.9.7 to 3.10.0rc2, which is due for final release *tomorrow*.

Works:

```sh
pyenv local 3.9.7
poetry env use $(pyenv which python)
poetry install
```

Fails:

```sh
pyenv local 3.10.0rc2
poetry env use $(pyenv which python)
poetry install
```

Poetry eats most of the stack trace. I see the following repeated once for each dependency:

```python
  • Installing urllib3 (1.26.7): Failed

  JSONDecodeError

  Expecting value: line 1 column 1 (char 0)

  at /usr/lib/python3.9/json/decoder.py:355 in raw_decode
      351│         """
      352│         try:
      353│             obj, end = self.scan_once(s, idx)
      354│         except StopIteration as err:
    → 355│             raise JSONDecodeError("Expecting value", s, err.value) from None
      356│         return obj, end
      357│ 
```

**Comment by laoshaw** [comment]
same issue with python 3.10-rc2 while 3.10 is supposed to be released today. python 3.9 worked well.

**Comment by playpauseandstop** [comment]
Hi all,

Wanted to update you that given error should gone with latest [`poetry==1.1.11`](https://github.com/python-poetry/poetry/releases/tag/1.1.11) release. With the release you don’t need to disable experimental new installer to install your dependencies with poetry at 3.10.0rc2 or 3.10.0.

**Comment by ksamuel** [comment]
Confirmed for me, fresh stable 3.10 works with latest poetry.

Be careful when upgrading, there are packages that don't have a wheel yet, and you'll need to compile them while in 3.9, you didn't need to.

For me, you can close the ticket.

**Comment by camerondavison** [comment]
@IceTDrinker same here for me. so confused how this could have just started happening.

**Comment by abn** [comment]
Can someone provide steps to reproduce this, trying to addd bs4 as dev dependency in a `python:3.8.12` container with `poetry@1.1.11` did not reproduce the issue. Ideally somehting I can reproduce inside a python container.

**Comment by IceTDrinker** [comment]
> Can someone provide steps to reproduce this, trying to addd bs4 as dev dependency in a `python:3.8.12` container with `poetry@1.1.11` did not reproduce the issue. Ideally somehting I can reproduce inside a python container.

I'll try to see if I can give you a minimal repro, but for me it started happening after the bs4 package was added and the cache from a previous github actions run was used.

**Comment by berislavlopac** [comment]
I've started getting this issue intermittently, when running `poetry update`, `add` or `install`. The commands fail on some of the dependencies, but not always the same; re-running the command usually results in one or two failures (with different packages each time), until it works usually on the third attempt or so.

**Comment by moriyoshi** [comment]
```
poetry cache clear --all pypi
poetry cache clear --all private
```
did the trick.  For me which python version it run on seemed irrelevant.

**Comment by s-nicholls** [comment]
@berislavlopac thank you! I re-ran a plugin add a few times and finally it worked, have the same JSON decode error usually when running:

poetry plugin add "keyrings.google-artifactregistry-auth"

**Comment by Sidiox** [comment]
Can reproduce on Windows.
Windows 10
Python 3.7.9
Poetry 1.1.11
pip 21.3.1

Previous workaround by reassigning the poetry cache `$env:POETRY_CACHE_DIR='.\poetrycache'` did not work.
Workaround clearing the cache by doing `poetry cache clear --all pypi` did seem to work.

**Comment by ddc67cd** [comment]
Try installing CacheControl==0.12.6 or wait for https://github.com/ionrock/cachecontrol/pull/264

**Comment by sblondon** [comment]
The error is resolved with `experimental.new-installer false` every time.  
With  `experimental.new-installer true`, sometimes the error `JSONDecodeError` is raised and it works in next attempt without changing `experimental.new-installer` value. I guess it matches with ddc67cd's [comment](https://github.com/python-poetry/poetry/issues/4210#issuecomment-956146936).

I tested with:
```
$ ./working_venv/bin/poetry config --list | grep new-installer
experimental.new-installer = true
```

```shell
#! /bin/sh

VENV_DIR="venv"
PYTHON='python3.9'

rm -rf "${VENV_DIR}"
${PYTHON} -m venv "${VENV_DIR}"

"./${VENV_DIR}/bin/pip" install --upgrade pip setuptools poetry

"./${VENV_DIR}/bin/poetry" i
```

`pyproject.toml` file:
```
[tool.poetry]
name = "demo"
version = "0.1.0"
description = "demo"
authors = ["<demo@domain.tld>"]

[tool.poetry.dependencies]
Markdown = "^3.*"

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

It can fail with other dependencies than Markdown.

There was a `poetry.toml` file during the tests but I don't think it's relevant:
```
[virtualenvs]
create = false
```

**Comment by ddc67cd** [comment]
So, the issue is resolved with the new version of cachecontrol==0.12.9 (it will be installed automatically).

Besides, when trying to reproduce this bug in a fresh container I see "SolverProblemError", thus, after this update some of the issues with that error could be resolved as well.

How to reproduce:
```
docker run --rm -ti python:3.9 bash
pip install poetry CacheControl==0.12.8
poetry init -n
poetry config experimental.new-installer false
poetry add asyncpg


wait 900 secs


poetry update  <- fails
poetry update  <- succees
```

**Comment by aalok-sathe** [comment]
I run into this issue on Python 3.8.8 as well.

**Comment by kolypto** [comment]
Here's how to fix it:

```shell
$ pip install -U cachecontrol
```

**Comment by emersonmx** [comment]
I use `rm -rf ~/.cache/pypoetry/cache/`.

**Comment by finswimmer** [comment]
Hello everyone,

as others already mention this issue is solved by a new version of `cachecontrol`.

If you install poetry by the [`install-poetry.py`](https://python-poetry.org/docs/master/#installation) scrip, the new version is installed automatically.

If you use the deprecated `get-poetry.py` script, it is not possible to update the `cachecontrol` version. `get-poetry.py` vendores the dependencies based on poetry's lock file. `get-poetry.py` installs poetry 1.1 which still supports python2. The current version of `cachecontrol` doesn't support python2 anymore, thus we cannot update the locked version in the 1.1 branch.

TL;DR;: Install poetry with the `install-poetry.py` to resolve this issue.

fin swimmer

**Comment by jcowles** [comment]
fresh install with install-poetry.py, deleting poetry.lock, and latest version of cachecontrol did not fix the issue for me, FYI

**Comment by jcowles** [comment]
The problem for me is that something was spewing "Cannot find the file specified" when launching python. Poetry initializes by running python and printing "sys.path" and then attempts to decode this using the json parser.

The JSON parser fails (loads) because the json string starts with "Cannot find the file specified", and of course that's not valid json.

Still a total mystery what is spewing that string though

**Comment by edmondo1984** [comment]
Installed poetry using pip on Windows on 3.9.6, cannot get poetry install to work :( Tried all suggesitons in this post, with no luck

**Comment by sandys** [comment]
is this bug fixed ? im running a fresh conda install on Windows WSL Ubuntu with Conda and Python 3.10 just now. I had the exact same issue. Only thing that worked was `poetry config experimental.new-installer false`

**Comment by operte** [comment]
I just had the same issue. I tried first deleting the virtual environment (as per @jamesowers-roo), but that didn't fix it. 

What worked was the solution from @hoefling:

> BTW disabling `poetry`'s experimental new installer may be a workaround for now:
> 
> ```shell
> $ poetry config experimental.new-installer false
> ```

**Comment by neersighted** [comment]
The JSONDecodeError discussed here is very specifically related to cache-control. Other JSONDecodeErrors are likely caused by broken systems/virtual environments and are more general (not Windows) versions of #6664.
